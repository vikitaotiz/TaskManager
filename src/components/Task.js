import { NavLink } from 'react-router-dom';

const Task = ({ task }) => {
  const { title, completed } = task;
  return (
    <div className="container mt-3">
      <div className="mb-4 d-flex justify-content-between">
        <NavLink to="/" className="text-decoration-none text-white">
          <i className="fas fa-arrow-left" />
          {' '}
          Back
        </NavLink>
        <b className="text-decoration-none text-white">Single Task</b>
        <span>
          {' '}
        </span>
      </div>
      <div className="card rounded-0 p-3 mb-2">
        {title}
      </div>
      <small className="card rounded-0 p-1">
        <i>{completed ? 'Completed' : 'Incomplete'}</i>
      </small>
    </div>
  );
};

export default Task;
