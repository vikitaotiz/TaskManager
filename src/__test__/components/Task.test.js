import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/storeConfig';
import Task from '../../components/Task';

it('should render correctly', () => {
  const body = document.createElement('tbody');
  const task = { id: 1, title: 'Task One', userId: 1 };

  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Task task={task} />
        </BrowserRouter>
      </Provider>,
      body,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
