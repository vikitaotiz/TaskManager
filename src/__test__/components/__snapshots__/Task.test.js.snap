// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="container mt-3"
>
  <div
    className="mb-4 d-flex justify-content-between"
  >
    <a
      aria-current="page"
      className="text-decoration-none text-white active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <i
        className="fas fa-arrow-left"
      />
       
      Back
    </a>
    <b
      className="text-decoration-none text-white"
    >
      Single Task
    </b>
    <span>
       
    </span>
  </div>
  <div
    className="card rounded-0 p-3 mb-2"
  >
    Task One
  </div>
  <small
    className="card rounded-0 p-1"
  >
    <i>
      Incomplete
    </i>
  </small>
</div>
`;
